api/controllers/search.js

```diff
@@ -32,14 +49,25 @@ const searchCollection = async function (roles, keywords, schemaName, pageNum, p
   // Create appropriate aggregations for the schema.
   let schemaAggregation;
   let matchAggregation;
+  let updatedIn30daysModifier;
   switch (schemaName) {
   case constants.DOCUMENT:
+    if (and && and.changedInLast30days) {
+      delete and.changedInLast30days;
+      updatedIn30daysModifier = documentAggregator.addUpdatedInLast30daysAggr();
+    }
     matchAggregation = await documentAggregator.createMatchAggr(schemaName, project, decodedKeywords, caseSensitive, or, and, categorized, roles, fuzzy);
     schemaAggregation = documentAggregator.createDocumentAggr(populate, roles, sortingValue, sortField, sortDirection, pageNum, pageSize);
+    if (updatedIn30daysModifier) schemaAggregation = [...schemaAggregation, ...updatedIn30daysModifier];
     break;
   case constants.PROJECT:
+    if (and && and.changedInLast30days) {
```

Comment:
Since this is just another matching condition, we can move it to the corresponding aggregator(documentAggregator or projectAggregator).